pkg/apis/core/validation/shoot.go

```diff
@@ -1220,6 +1220,18 @@ func ValidateVerticalPodAutoscaler(autoScaler core.VerticalPodAutoscaler, fldPat
 	if percentile := autoScaler.RecommendationUpperBoundMemoryPercentile; percentile != nil {
 		allErrs = append(allErrs, validatePercentile(*percentile, fldPath.Child("recommendationUpperBoundMemoryPercentile"))...)
 	}
+	if halfLife := autoScaler.CPUHistogramDecayHalfLife; halfLife != nil && halfLife.Duration < 0 {
```

Comment:
Nit: Is it possible to use `ValidatePositiveDuration`?

Reply:
I was just about to open a separate issue about that. It seems that we are using different approaches when validating that `time.Duration` should be bigger than 0 -  some places in the code use `ValidatePositiveDuration`, some places use `apivalidation.ValidateNonnegativeField(int64(<time.Duration>), fldPath.Child("<field>"))...)` and some places use the same if as the current code.
We should unify all of those, but perhaps in a separate PR.

Reply:
Oh, I completely missed this, thanks!